flink-formats/flink-parquet/src/test/java/org/apache/flink/formats/parquet/ParquetColumnarRowInputFormatTest.java

```diff
@@ -294,6 +294,80 @@ void testProjectionReadUnknownField(int rowGroupSize) throws IOException {
                 });
     }
 
+    @ParameterizedTest
+    @MethodSource("parameters")
+    void testProjectionReadUnknownFieldAcrossFiles(int rowGroupSize) throws IOException {
+        int number = 1000;
+        List<RowData> records = new ArrayList<>(number);
+        for (int i = 0; i < number; i++) {
+            Integer v = i;
+            records.add(newRow(v));
+        }
+        // create parquet files for both legacy and new
+        // f7 and f99 don’t exist in legacy parquet file
+        // f99 doesn't exist in new parquet file
+        // create legacy parquet file
+        RowType legacyParquetType =
+                new RowType(
+                        IntStream.range(0, 6) // assume f7 not exist in legacy parquet file.
+                                .mapToObj(i -> ROW_TYPE.getFields().get(i))
+                                .collect(Collectors.toList()));
+        Path legacyParquetPath =
+                createTempParquetFile(
+                        new File(folder, "legacy"), legacyParquetType, records, rowGroupSize);
+        // create new parquet file
+        Path newParquetPath = createTempParquetFile(new File(folder, "new"), records, rowGroupSize);
+
+        // test reader
+        String[] readerColumnNames = new String[] {"f7", "f2", "f4", "f99"};
+        LogicalType[] fieldTypes =
+                new LogicalType[] {
+                    new DoubleType(), new TinyIntType(), new IntType(), new VarCharType()
+                };
```

Comment:
```suggestion
        final LogicalType[] fieldTypes = Arrays.stream(readerColumnNames)
                        .map(ROW_TYPE::getFieldIndex)
                        .map(ROW_TYPE::getTypeAt)
                        .toArray(LogicalType[]::new);
```
Could we use that instead? This would make the test more robust against test data changes (i.e. the data types are dynamically derived from the static `ROW_TYPE` instead of hard-coded). WDYT?